src/lib/components/Request.svelte

```diff
@@ -1,60 +1,78 @@
 <script lang="ts">
   import { sendRequest } from "../handlers/socketio";
-  import { serverSettings,request } from "../store";
+  import { serverSettings, request } from "../store";
   import ConnectionController from "./ConnectionController.svelte";
   import ServerAddressModal from "./ServerAddressModal.svelte";
-  import { getNotificationsContext } from 'svelte-notifications';
+  import { getNotificationsContext } from "svelte-notifications";
   import { nanoid } from "nanoid";
   const { addNotification } = getNotificationsContext();
 
-  function requestHandler(){
-    if($serverSettings.status !== 'connected'){
+  let tabAsSpaces = true;
+  let statusMessage = "";
+
+  function requestHandler() {
+    if ($serverSettings.status !== "connected") {
       addNotification({
-            text: 'Server is not connected',
-            position: 'bottom-center',
-            type: 'danger',
-            removeAfter: 3000,
-        })
+        text: "Server is not connected",
+        position: "bottom-center",
+        type: "danger",
+        removeAfter: 3000,
+      });
       return;
     }
-    if(!$request.emitName || $request.emitName.length === 0){
+    if (!$request.emitName || $request.emitName.length === 0) {
       addNotification({
-            text: 'Emit name is not set',
-            position: 'bottom-center',
-            type: 'danger',
-            removeAfter: 3000,
-        })
+        text: "Emit name is not set",
+        position: "bottom-center",
+        type: "danger",
+        removeAfter: 3000,
+      });
       return;
     }
 
-    try{
-      if(!$request.title) $request.title = nanoid(4);
+    try {
+      if (!$request.title) $request.title = nanoid(4);
       sendRequest();
-    }catch(e){
+    } catch (e) {
       addNotification({
-            text: e,
-            position: 'bottom-center',
-            type: 'danger',
-            description: e,
-            removeAfter: 3000,
-        })
+        text: e,
+        position: "bottom-center",
+        type: "danger",
+        description: e,
+        removeAfter: 3000,
+      });
     }
   }
 
-  function handleTextArea(e){
-    if(e.code == 'Tab') {e.preventDefault()
-      $request.body = $request.body + '    ';
-    };
+  function handleTextArea(e) {
+    if (e.code == "Tab" && tabAsSpaces) {
+      e.preventDefault();
+      $request.body = $request.body + "    ";
+    } else if (e.code == "KeyM" && e.ctrlKey) {
```

Comment:
I think it would be better to revert this action once the user starts typing in the textarea. It might be confusing if the user forget to press ctrl +m

Reply:
I agree with you about taht, the thing is that a keyboard user needs to tab to get to the buttons but if tab is blocked and inserts spaces, they (we) can not. I don't have another idea in how to solve that. DO you have any suggestion?

Reply:
I think it can be a good idea to place a checkbox on top of the textarea to let the user activate tap mode on that one.